cli/flox-rust-sdk/src/models/manifest/composite.rs

```diff
@@ -0,0 +1,276 @@
+#![allow(dead_code)] // TODO: Remove on first use.
+
+use std::iter::once;
+
+use flox_core::Version;
+use thiserror::Error;
+
+use super::typed::{
+    Manifest,
+    ManifestBuild,
+    ManifestContainerize,
+    ManifestHook,
+    ManifestInstall,
+    ManifestOptions,
+    ManifestProfile,
+    ManifestServices,
+    ManifestVariables,
+};
+
+#[derive(Error, Debug)]
+pub enum MergeError {}
+
+/// A collection of manifests to be merged with a `ManifestMergeStrategy`.
+struct CompositeManifest {
```

Comment:
Suggestion: In the ticket, the proposed implementation has this container implement the trait. The way it's currently written you have a `merge` method on the trait and a `merge` method on this struct, which is not the same as the trait method. I'd prefer if they weren't duplicated, and I think it makes sense for it to live on the trait implementor.

Reply:
I liked Matthew's proposal here: https://github.com/flox/flox/issues/2664#issuecomment-2623165797

We could give the methods different names (any suggestions?) but I don't feel like they're duplication since `CompositeManifest::merge` is applying any strategy to many manifests and `ManifestMergeStrategy::merge` is applying a given strategy to two manifests.

Reply:
On second thought I think I prefer this approach for the following reasons (this is mostly restating what you just said):
- The merge strategy and the merge application are separate.
- The merge strategy decides how two manifests are merged.
- The merge application decides which manifests are merged and in what order.

I do prefer the `merge` methods having different names, otherwise it will be misleading at the call site. Maybe `merge_all` for the one currently on `CompositeManifest`.

Reply:
> although this now requires that the merge strategy have zero knowledge about the two manifests that it's merging e.g. if it wants to treat the composing manifest different it now has no way to do that 🤷‍♂️

Good point, and we may need to change this when implement "overrides", but we can deal with that when it comes.